includes/class-wc-checkout.php

```diff
@@ -825,12 +827,20 @@ protected function validate_posted_data( &$data, &$errors ) {
 	 * @since  3.0.0
 	 * @param  array    $data   An array of posted data.
 	 * @param  WP_Error $errors Validation errors.
+	 * @throws Exception Invalid country code in billing or shipping address.
 	 */
 	protected function validate_checkout( &$data, &$errors ) {
 		$this->validate_posted_data( $data, $errors );
 		$this->check_cart_items();
 
-		if ( empty( $data['woocommerce_checkout_update_totals'] ) && empty( $data['terms'] ) && ! empty( $_POST['terms-field'] ) ) { // WPCS: input var ok, CSRF ok.
+		$billing_country = WC()->customer->get_shipping_country();
```

Comment:
Looking at line https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-checkout.php#L759 it seems like this method is used as a fallback and not as default way to get country address (unless cart needs shipping). Perhaps we should move these checks to inside `validate_posted_data` method only if country is a required field (it might not be for some shops, I think looking at the linked code).

Reply:
I see an inconsistency here. On one hand the `validate_posted_data` method is doing

```php
$country = isset( $data[ $fieldset_key . '_country' ] ) ? $data[ $fieldset_key . '_country' ] : WC()->customer->{"get_{$fieldset_key}_country"}();
```

but on the other hand the `validate_checkout` method does just

```php
if ( WC()->cart->needs_shipping() ) {
	$shipping_country = WC()->customer->get_shipping_country();
```

so maybe that needs fixing too?
```

Reply:
After a second look I see that `WC()->customer->get_shipping_country()` is effectively getting the country from the request... not sure if that's intentional (it's confusing at least).